docs/develop/go/test-suites.mdx

```diff
@@ -160,6 +161,103 @@ When an Activity sends a Heartbeat, be sure that you can see the Heartbeats in y
+it belongs to the Nexus service `sample-service`.
+
+```go
+func (s *UnitTestSuite) Test_SimpleWorkflow_NexusSyncOperation() {
+        s.env.OnNexusOperation(
+                "sample-service",
+                nexus.NewOperationReference[SampleInput, SampleOutput]("sample-operation"),
+                SampleInput{},
+                workflow.NexusOperationOptions{},
+        ).Return(
+                &nexus.HandlerStartOperationResultSync[SampleOutput]{
+                        Value: SampleOutput{},
+                },
+                nil, // error if you want to simulate an error in the ExecuteOperation call
+        )
+        // You can also add a delay to return the mock values by calling After().
+        // Eg: s.env.OnNexusOperation(...).Return(...).After(1*time.Second)
+
+        s.env.ExecuteWorkflow(SimpleWorkflow, "test_nexus_operation")
+
+        s.True(s.env.IsWorkflowCompleted())
+        s.NoError(s.env.GetWorkflowError())
+}
+```
+
+The example above simulates a call to a Nexus synchronous operation, indicated by the returned value
+type `*nexus.HandlerStartOperationResultSync[T]`. Since `OnNexusOperation` needs to know the
+operation's name, input and output, `nexus.NewOperationReference` creates an operation reference
+that represents the operation without knowing the implementation (basically, it represents the
+signature of the operation). You may also use the operation itself instead of creating the operation
+reference if you have it available.
+
+Let's take an example of simulating a call to an Nexus asynchronous operation.
+
+```go
+func (s *UnitTestSuite) Test_SimpleWorkflow_NexusAsyncOperation() {
+        s.env.OnNexusOperation(
+                "sample-service",
+                nexus.NewOperationReference[SampleInput, SampleOutput]("sample-operation"),
+                SampleInput{},
+                workflow.NexusOperationOptions{},
+        ).Return(
+                &nexus.HandlerStartOperationResultAsync{
+                        OperationID: "sample-operation-id",
+                },
+                nil, // error if you want to simulate an error in the ExecuteOperation call
+        )
+        err := env.RegisterNexusAsyncOperationCompletion(
+                "sample-service",
+                "sample-operation",
+                "sample-operation-id", // must match the OperationID above
+                SampleOutput{},
+                nil,            // error if you want to simulation an error in the operation
+                2*time.Second,  // delay to simulate how long the operation takes after it starts
+        )
+        s.NoError(err)
+        s.env.ExecuteWorkflow(SimpleWorkflow, "test_nexus_operation")
+
+        s.True(s.env.IsWorkflowCompleted())
+        s.NoError(s.env.GetWorkflowError())
+}
+```
+
+In the example above, the `OperationID` in the `*nexus.HandlerStartOperationResultAsync` object can
```

Comment:
My quick read-through doesn't show multiple asynchronous calls, so maybe that should be a separate paragraph, and any details on how you'd do that?

Reply:
I just meant that the user can mock more than one Nexus async operation with different `OperationID`, and they would have to call `RegisterNexusAsyncOperationCompletion` for each of them with the corresponding `OperationID`.